docs/en/10-public-api.md

```diff
@@ -684,4 +684,269 @@ Example successful response:
     "time": "2024-12-23T18:23:41.349Z"
   }
 }
-```
\ No newline at end of file
+```
+
+## Async search gRPC API
+
+### `/StartAsyncSearch`
+
+Start async search.
+
+Returns id that is later can be used to fetch a result, cancel and delete search.
+
+Similar to [`/ComplexSearch`](#complexsearch) but has additional fields `retention` and `with_docs`:
+`retention` determines how long the data will be available.
+And `with_docs` determines whether documents ids will be found and saved.
+`with_total` will automatically be set to true when `with_docs` is true.
+
+Example request:
+
+```bash
+grpcurl -plaintext -d '
+{
+  "retention": "3600s",
+  "query": {
+    "from": "2025-07-01T05:20:00Z",
+    "to": "2025-09-01T05:21:00Z",
+    "query": "message:error | fields level, message"
+  },
+  "hist": {
+    "interval": "1ms"
+  },
+  "aggs": [
+    {
+      "group_by": "k8s_pod",
+      "field": "request_time",
+      "func": "AGG_FUNC_QUANTILE",
+      "quantiles": [
+        0.2,
+        0.8,
+        0.95
+      ]
+    }
+  ],
+  "with_docs": true
+}}' localhost:9004 seqproxyapi.v1.SeqProxyApi/StartAsyncSearch
+```
+
+Example successful response:
+
+```json
+{
+  "search_id": "c28c97d1-117a-45dc-a0cf-d080f22b2a10"
+}
+```
+
+### `/FetchAsyncSearchResult`
+
+Fetch async search result.
+Returns execution result based on a search parameters from [`/StartAsyncSearch`](#startasyncsearch) request
```

Comment:
maybe we should mention that an this request can return a partial response, depending on whether the search is done or not. and also mention the status field